viper.py

```diff
@@ -172,115 +168,113 @@ def SSRstat(vgrid, SSR, dk=1, plot='maybe', N=None):
-    x_ok = x[i_ok]
-    w_ok = w[i_ok]
-    f_ok = f[i_ok]
+    # select good pixel
+    i_ok = np.where(flag_pixel==0)[0]
+    pixel_ok = pixel[i_ok]
+    wave_obs_ok = wave_obs[i_ok]
+    spec_obs_ok = spec_obs[i_ok]
 
-    modset['icen'] = icen = np.nanmean(x_ok) + 18   # slight offset, then it converges for CES+TauCet
+    modset['icen'] = icen = np.nanmean(pixel_ok) + 18   # slight offset, then it converges for CES+TauCet
     modset['IP_hs'] = iphs
 
-    if degar:
+    if deg_norm_rat:
         # rational polynomial
-        modset['envelope'] = lambda x, a: pade(x, a[:dega+1], a[dega+1:])
-
-    # display
-    # pre-look raw input
-    s = slice(*np.searchsorted(w_I2, [lmin, lmax]))
-    s_s = slice(*np.searchsorted(w_tpl[o], [lmin, lmax]))
-    sj = slice(*np.searchsorted(uj_full, np.log([lmin, lmax])))
-
-    # prepare input; convert discrete data to model
-
-    # using the supersampled log(wavelength) space with knot index j
-    uj = uj_full[sj]
-    iod_j = iod_j_full[sj]
+        modset['envelope'] = lambda x, par_norm: pade(x, par_norm[:deg_norm+1], par_norm[deg_norm+1:])
 
     if telluric == 'add':
-        atmj = np.zeros((0,len(uj)))
-        for at in range(0,len(molec),1):
-            s_t = slice(*np.searchsorted(w_atm[at], [lmin, lmax]))
+        # select present molecules for telluric forward modeling
+        spec_atm = np.zeros((0,len(lnwave_cell)))
+        for mol in range(0,len(molec),1):
+            s_atm = slice(*np.searchsorted(wave_atm_all[mol], [lmin, lmax]))
             # bring it to same log(wavelength) scale as cell
-            atmi = np.interp(uj, np.log(w_atm[at][s_t]), f_atm[at][s_t])
+            atmi = np.interp(lnwave_cell, np.log(wave_atm_all[mol][s_atm]), spec_atm_all[mol][s_atm])
             # chose just present molecules in wavelength range
             if np.nanstd(atmi) > 0.0001:
-                atmj = np.r_[atmj,[atmi]]
+                spec_atm = np.r_[spec_atm,[atmi]]
         # parameter to scale each telluric model:
 	# add parameter for telluric position shift if selected
-        t = t0 = np.ones(len(atmj)+int(tellshift))
+        par_atm = par_atm0 = np.ones(len(spec_atm)+int(tellshift))
     else:
-        atmj = []
-        t = t0 = []
+        spec_atm = []
+        par_atm = par_atm0 = []
+
 
     if demo & 1:
+        # pre-look raw input
+        s_cell = slice(*np.searchsorted(wave_cell, [lmin, lmax]))
+        s_tpl = slice(*np.searchsorted(wave_tpl[order], [lmin, lmax]))
+
         # plot data, template, and iodine with some scaling
-        gplot(w_I2[s], f_I2[s]/np.nanmedian(f_I2[s]), 'w l lc 9,', w_tpl[o][s_s], f_tpl[o][s_s]/np.nanmedian(f_tpl[o][s_s]), 'w l lc 3,', w, f/np.nanmedian(f), 'w lp lc 1 pt 7 ps 0.5')
+        gplot(wave_cell[s_cell], spec_cell[s_cell]/np.nanmedian(spec_cell[s_cell]), 'w l lc 9,', wave_tpl[order][s_tpl], spec_tpl[order][s_tpl]/np.nanmedian(spec_tpl[order][s_tpl]), 'w l lc 3,', wave_obs, spec_obs/np.nanmedian(spec_obs), 'w lp lc 1 pt 7 ps 0.5')
```

Comment:
Was thinking about that for a while. Same for `nanmean`. Will do it one day, when I have too much time. This will take a bit of time and will be no fun changing this in the entire code.